spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaMessageListenerContainerTests.java

```diff
@@ -560,6 +564,67 @@ public void testBatchAck() throws Exception {
 		logger.info("Stop batch ack");
 	}
 
+	@Test
+	public void testBatchListener() throws Exception {
+		logger.info("Start batch listener");
+
+		Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+		ProducerFactory<Integer, String> pf = new DefaultKafkaProducerFactory<>(senderProps);
+		KafkaTemplate<Integer, String> template = new KafkaTemplate<>(pf);
+		template.setDefaultTopic(topic8);
+		template.sendDefault(0, 0, "foo");
+		template.sendDefault(0, 0, "baz");
+		template.sendDefault(1, 0, "bar");
+		template.sendDefault(1, 0, "qux");
+		template.flush();
+
+		Map<String, Object> props = KafkaTestUtils.consumerProps("test8", "false", embeddedKafka);
+		props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+		DefaultKafkaConsumerFactory<Integer, String> cf = new DefaultKafkaConsumerFactory<>(props);
+		ContainerProperties containerProps = new ContainerProperties(topic8);
+		containerProps.setMessageListener((BatchMessageListener<Integer, String>) message -> {
+			logger.info("batch listener: " + message);
```

Comment:
That's why I added `topic9` but was too lazy to do a manual version of this test (which used `AckMode.BATCH`).